### PR TITLE
support '@' completions via .AtNames

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/13451.

### Approach

If `.AtNames` is defined and available, use it when requesting completions following an `@`.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/13451.

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Review by jgutman — DISMISSED** [review]
Looks good! How important is this -- should this be backported?

**Comment by kevinushey** [comment]
I don't think it's necessary to backport; this just adds support for a new completion option introduced in R 4.3.0.

**Comment by jgutman** [comment]
@kevinushey it looks like they also added similar support for `.DollarNames`.

```
`$.myclass` <- function(x, name) {
   attr(x, name)
 }
 
 DollarNames.myclass <- function(x, pattern) {
   utils::findMatches(pattern, names(attributes(x)))
}

x <- structure(list(), class = "myclass", attr1 = "attr1val")

x$
```
i.e. this tab completes in R but not in RStudio and likely needs a similar addition

File: src/cpp/session/modules/SessionRCompletions.R [hunk 1]
```diff
@@ -1526,16 +1526,30 @@ assign(x = ".rs.acCompletionTypes",
...
+            code <- substitute(
+               utils::.AtNames(object, pattern = ""),
+               list(object = object)
+            )
+            allNames <- eval(code, envir = globalenv())
+            names <- .rs.selectFuzzyMatches(allNames, token)
+            type <- attr(names, "types")
+         })
+      }
+      else if (isS4(object))
```
Comment (jgutman) [inline src/cpp/session/modules/SessionRCompletions.R:1541]: Actually on second thought I don't get why this is an else if? So we only do the s4 slot completions on older R versions where AtNames is not defined?

Comment (jgutman) [reply src/cpp/session/modules/SessionRCompletions.R:1541]: Oh I see now that S4 objects also use the .AtNames so this works for standard S4 object definitions too

**Review by jgutman — APPROVED** [review]
LGTM, but can we add the `.DollarNames` autocompletion as well?

**Pull request merged**